### PR TITLE
Floating-point numbers comparison util for tests and benchmarks

### PR DESCRIPTION
**Pull request by Nyrio**

### CODE REVIEW

**Comment by mehdi-goli** [comment]
LGTM!

File: include/utils/float_comparison.hpp [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+  return static_cast<scalar_t>(0.005);
+}
+
+template <>
+inline double getRelativeErrorMargin<double>() {
+  return 0.000001;
+}
+
+/**
+ * Indicates the tolerated margin for absolute differences
```
Comment (mehdi-goli) [inline include/utils/float_comparison.hpp]: Put a comment and explain how did you came up with these numbers?

Comment (Nyrio) [reply include/utils/float_comparison.hpp]: I've added a few comments

**Comment by Nyrio** [comment]
@mehdi-goli it's only for floating-point numbers. For integers an exact match can be expected (e.g for iamin and iamax I check that the index is the exact same)

File: include/utils/float_comparison.hpp (hunk 1 above)
Comment (Nyrio) [reply include/utils/float_comparison.hpp]: I have added even more explanations. I hope it's clear.
Also I've made a change to properly handle the case of zero

**Pull request merged**